fix: 15963 - Border line appears, then disappears, above the 1st search result

tienifr opened the pull request:
<!-- If necessary, assign reviewers that know the area or changes well. Feel free to tag any additional reviewers you see fit. -->

### Details
<!-- Explanation of the change or anything fishy that is going on -->

This PR added `this.props.shouldHaveOptionSeparator !== nextProps.shouldHaveOptionSeparator;` to the `shouldComponentUpdate` of `OptionRow`. This will remove the redundant border line above the first item in the pronoun search.

In addition, the PR also added some other props condition in `shouldComponentUpdate` that should make the item in `OptionRow` re-render, as suggested by the reviewer. The addition props being considered are `this.props.option.shouldShowSubscript`, `this.props.option.ownerEmail`, `this.props.option.subtitle`, `this.props.option.pendingAction`.

### Fixed Issues
<!---
1. Please replace GH_LINK with a URL link to the GitHub issue this Pull Request is fixing.
2. Please replace PROPOSAL: GH_LINK_ISSUE(COMMENT) with a URL link to your GitHub comment, which contains the approved proposal (i.e. the proposal that was approved by Expensify).

Do NOT add the special GH keywords like `fixed` etc, we have our own process of managing the flow.
It MUST be an entire link to the github issue and your comment proposal ; otherwise, the linking will not work as expected.

Make sure this section looks similar to this (you can link multiple issues using the same formatting, just add a new line):

$ https://github.com/Expensify/App/issues/<number-of-the-issue>
$ https://github.com/Expensify/App/issues/<number-of-the-issue(comment)>

Do NOT only link the issue number like this: $ #<number-of-the-issue>
--->
$ https://github.com/Expensify/App/issues/15963
PROPOSAL: https://github.com/Expensify/App/issues/15963#issuecomment-1469410712


### Tests
<!---
Add a numbered list of manual tests you performed that validates your changes work on all platforms, and that there are no regressions present.
Add any additional test steps if test steps are unique to a particular platform.
Manual test steps should be written so that your reviewer can repeat and verify one or more expected outcomes in the development environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->

1. Navigate to Settings > Profile > Pronouns
2. Type Hi
3. Observe that the border doesn't show above the first result
4. Type Hir
5. Verify that the border still doesn't show above the first result

- [x] Verify that no errors appear in the JS console

### Offline tests
<!---
Add any relevant steps that validate your changes work as expected in a variety of network states e.g. "offline", "spotty connection", "slow internet", etc. Manual test steps should be written so that your reviewer and QA testers can repeat and verify one or more expected outcomes. If you are unsure how the behavior should work ask for advice in the `#expensify-open-source` Slack channel.
--->

n/a

### QA Steps
<!---
Add a numbered list of manual tests that can be performed by our QA engineers on the staging environment to validate that your changes work on all platforms, and that there are no regressions present.
Add any additional QA steps if test steps are unique to a particular platform.
Manual test steps should be written so that the QA engineer can repeat and verify one or more expected outcomes in the staging environment.

For example:
1. Click on the text input to bring it into focus
9. Upload an image via copy paste
10. Verify a modal appears displaying a preview of that image
--->

#### For the main issue
1. Navigate to Settings > Profile > Pronouns
2. Type Hi
3. Observe that the border doesn't show above the first result
4. Type Hir
5. Verify that the border still doesn't show above the first result


#### For the other pages that could be affected

##### PriorityModePage
1. Navigate to Settings > Preferences > Priority mode
6. Verify that the border doesn't show above the first item
7. Verify that there is no error occur

##### LanguagePage
1. Navigate to Settings > Preferences > Language
2. Verify that the border doesn't show above the first item
3. Verify that there is no error occur

##### ReportParticipantsPage
1. Navigate to a report with multiple participants
2. Click on the user list on the top to open Details page
3. Verify that the border doesn't show above the first item
8. Verify that there is no error occur

##### IOUConfirmationList
1. Go to a report
2. Choose `Request money` or `Send money`
3. Type in an amount and move to the confirmation page
4. Verify that there is no error occur

##### NewChatPage
1. Click the plus button in the LHN page, select `New chat`
2. Verify that the border doesn't show above the first item
3. Verify that there is no error occur

##### SearchPage
1. Click the search button in the LHN page
2. Verify that the border doesn't show above the first item
3. Verify that there is no error occur

##### IOUCurrencySelection
1. Go to a report
2. Choose `Request money` or `Send money`
3. Select the currency to enter currency selection page
4. Verify that the border doesn't show above the first item
9. Verify that there is no error occur

##### IOUParticipantsRequest
1. Click the plus button in the LHN page, select `Request money`
2. Type in an amount and proceed
3. Search for users
4. Verify that the border doesn't show above the first item
5. Verify that there is no error occur

##### IOUParticipantsSplit
1. Click the plus button in the LHN page, select `Split bill`
2. Type in an amount and proceed
3. Search for users
4. Verify that the border doesn't show above the first item
5. Verify that there is no error occur

##### YearPickerPage
1. Navigate to Settings > Profile > Personal details > Date of birth
2. Select year dropdown
3. Search for a year
4. Verify that the border doesn't show above the first item
5. Verify that there is no error occur

##### TimezoneSelectPage
1. Navigate to Settings > Profile > Timezone
2. Disable automatic timezone, then go to select time zone
3. Verify that the border doesn't show above the first item
4. Verify that there is no error occur

##### WorkspaceInvitePage
1. Go to a workspace
2. Select `Manage members` > `Invite`
3. Search for a user
4. Verify that the border doesn't show above the first item
5. Verify that there is no error occur

##### WorkspaceMembersPage
1. Go to a workspace
2. Select `Manage members`
3. Verify that the border doesn't show above the first item
4. Verify that there is no error occur

- [x] Verify that no errors appear in the JS console

### PR Author Checklist
<!--
This is a checklist for PR authors. Please make sure to complete all tasks and check them off once you do, or else your PR will not be merged!
-->

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://user-images.githubusercontent.com/113963320/226094178-18db0b8d-a574-427a-9ab7-7c86f9afc6da.mov

https://user-images.githubusercontent.com/113963320/226094181-79b6ad33-89b5-4872-aec7-024aa38af7ad.mov

<!-- add screenshots or videos here -->

</details>

<details>
<summary>Mobile Web - Chrome</summary>

[android-chrome.webm](https://user-images.githubusercontent.com/113963320/226094175-0109103f-c9ad-4c0a-9630-987aa17c62b0.webm)

<!-- add screenshots or videos here -->

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://user-images.githubusercontent.com/113963320/226094171-2f8109e5-439b-487c-93c1-f25f4b001fe2.mov


<!-- add screenshots or videos here -->

</details>

<details>
<summary>Desktop</summary>

https://user-images.githubusercontent.com/113963320/226094148-b2e15884-b481-47f3-b816-15b9251eca08.mov

<!-- add screenshots or videos here -->

</details>


<details>
<summary>iOS</summary>


https://user-images.githubusercontent.com/113963320/226094647-0040ef39-5b86-478e-bf87-7b263eb0372e.mov


<!-- add screenshots or videos here -->

</details>

<details>
<summary>Android</summary>

[android-native.webm](https://user-images.githubusercontent.com/113963320/226094132-f5eb52ec-41d6-413e-9075-1e8865815da8.webm)

<!-- add screenshots or videos here -->

</details>

tienifr committed:
fix 15963: Add condition for border line display

tienifr committed:
fix: Add condition for border line display

s77rt reviewed (changes requested):
Can you please check for other props that we should re-render the item if they got changed e.g. `shouldDisableRowInnerPadding`, `option.isPinned`, `option.pendingAction`,  etc. Just add those that are likely to be changed.

tienifr committed:
Additional checks in OptionRow shouldComponentUpdate

tienifr committed:
Merge branch 'fix/15963' of https://github.com/tienifr/App into fix/15963

tienifr commented:
@s77rt 

> `shouldDisableRowInnerPadding`, `option.isPinned`, `option.pendingAction`

`shouldDisableRowInnerPadding` is not necessarily to be change, since it’s used as a fixed boolean props with the OptionList component [here](https://github.com/Expensify/App/blob/b82ef1e9adb11e75e8bf4b032c36d4efe0552c3d/src/pages/settings/Preferences/LanguagePage.js#L60) and [here](https://github.com/Expensify/App/blob/893d07a3d460bfd6e1210f562f531d2a8037469a/src/pages/settings/Preferences/PriorityModePage.js#L66). Similar thing goes for `props.showTitleTooltip`

`option.isPinned` is not used in `OptionRow`, so I think it wouldn’t be necessary to check it in `shouldComponentUpdate`.

Some attributes of `props.option` are added, namely `shouldShowSubscript`, `icons`, `ownerEmail`, `subtitle`, `pendingAction`. I agree that they should be presented since they're used in [OptionRow](https://github.com/Expensify/App/blob/5a0c876fabbccf8279eb5483c92997f565633267/src/components/OptionRow.js#L148). Their changes could affect the behavior of the `render` function.

Thoughts?

s77rt commented on src/components/OptionRow.js:
```diff
@@ -100,7 +100,13 @@ class OptionRow extends Component {
-            || this.props.option.brickRoadIndicator !== nextProps.option.brickRoadIndicator;
+            || this.props.option.brickRoadIndicator !== nextProps.option.brickRoadIndicator
+            || this.props.option.shouldShowSubscript !== nextProps.option.shouldShowSubscript
+            || this.props.option.icons !== nextProps.option.icons
```
```suggestion
```
We already check for icons.

s77rt commented on src/components/OptionRow.js:
```diff
@@ -100,7 +100,13 @@ class OptionRow extends Component {
+            || this.props.option.ownerEmail !== nextProps.option.ownerEmail
+            || this.props.option.subtitle !== nextProps.option.subtitle
+            || this.props.option.pendingAction !== nextProps.option.pendingAction
+            || this.props.shouldHaveOptionSeparator !== nextProps.shouldHaveOptionSeparator;
```
Move `props.shouldHaveOptionSeparator` check to be after `props.isSelected` since it's more likely to change compared to others (early return optimisation)

s77rt reviewed (changes requested):
Thanks for the update. I have requested some minor changes. Also I think we can also add `allReportErrors` but not sure what cases this would cover as I'd imagine `pendingAction` will change too most of the time we get an error, so let's skip it for now since it's an object that is not easy to compare and no clear purpose is identified.

tienifr committed:
fix: optimize code

tienifr commented:
@s77rt Thanks for pointing that out, I've removed icons checking and moved `props.shouldHaveOptionSeparator` check to be after `props.isSelected`. Please take a look

s77rt commented:
Also please remove the offline tests; write "n/a". I don't think offline tests have any different behavior then being online.

tienifr commented:
@s77rt Updated as suggested 🎉 Thanks!

s77rt commented:
## Reviewer Checklist

- [X] I have verified the author checklist is complete (all boxes are checked off).
- [X] I verified the correct issue is linked in the `### Fixed Issues` section above
- [X] I verified testing steps are clear and they cover the changes made in this PR
    - [X] I verified the steps for local testing are in the `Tests` section
    - [X] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [X] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [X] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [X] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [X] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [X] I verified tests pass on **all platforms** & I tested again on:
    - [X] Android / native
    - [X] Android / Chrome
    - [X] iOS / native
    - [X] iOS / Safari
    - [X] MacOS / Chrome / Safari
    - [X] MacOS / Desktop
- [X] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [X] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [X] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [X] I verified that comments were added to code that is not self explanatory
    - [X] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [X] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [X] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [X] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [X] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [X] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [X] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [X] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [X] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [X] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [X] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [X] If a new component is created I verified that:
    - [X] A similar component doesn't exist in the codebase
    - [X] All props are defined accurately and each prop has a `/** comment above it */`
    - [X] The file is named correctly
    - [X] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [X] The only data being stored in the state is data necessary for rendering and nothing else
    - [X] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [X] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [X] All JSX used for rendering exists in the render method
    - [X] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [X] If any new file was added I verified that:
    - [X] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [X] If a new CSS style is added I verified that:
    - [X] A similar style doesn't already exist
    - [X] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [X] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [X] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [X] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [X] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [X] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>


https://user-images.githubusercontent.com/16493223/226663287-d4222e27-f564-4542-92a1-a6dbd322eaf4.mp4


</details>

<details>
<summary>Mobile Web - Chrome</summary>


https://user-images.githubusercontent.com/16493223/226663144-7dc0bdd0-95a4-438e-9ecc-4c7593e3ad5c.mp4


</details>

<details>
<summary>Mobile Web - Safari</summary>

https://user-images.githubusercontent.com/16493223/226663094-d84e818b-1635-49af-91ac-dbbafd7613fb.mp4



</details>

<details>
<summary>Desktop</summary>

https://user-images.githubusercontent.com/16493223/226662786-ef461340-2f9a-42fc-ada0-07762513aad1.mp4


</details>

<details>
<summary>iOS</summary>


https://user-images.githubusercontent.com/16493223/226662699-b87b6c4f-5115-41b9-9f3e-398b6f9cf6ff.mp4


</details>

<details>
<summary>Android</summary>


https://user-images.githubusercontent.com/16493223/226662628-b201574a-32a2-4add-a2c8-03a49765848e.mp4


</details>

s77rt reviewed (approved):
LGTM! :rocket: 

cc @Beamanator

tienifr commented:
@Beamanator Could you help to check this PR?

tienifr commented:
@Beamanator Can you assist me in reviewing this PR?

s77rt commented:
@tienifr No need for double-tag. I'm sure Alex will check this once he get the time.

Beamanator commented:
Ya sorry for the delay, have been quite busy and I'm mainly on vacation this week but I'll be happy to merge soon, but I do have one question:

@s77rt @tienifr did y'all test this change on all pages that have the OptionRow? We don't want to have any regressions, and right now in the QA steps I only see tests for the Pronouns page

s77rt commented:
@Beamanator I confirm OptionRow is working fine on:
- PronounsPage
- PriorityModePage
- LanguagePage
- ReportParticipantsPage
- IOUConfirmationList
- NewChatPage
- SearchPage
- YearPickerPage
- IOUCurrencySelection
- IOUParticipantsRequest
- IOUParticipantsSplit
- TimezoneSelectPage
- WorkspaceInvitePage
- WorkspaceMembersPage


@tienifr Can you please confirm the same

tienifr commented:
@s77rt @Beamanator It works for me in all places

- [x] PronounsPage
- [x] PriorityModePage
- [x] LanguagePage
- [x] ReportParticipantsPage
- [x] IOUConfirmationList
- [x] NewChatPage
- [x] SearchPage
- [x] YearPickerPage
- [x] IOUCurrencySelection
- [x] IOUParticipantsRequest
- [x] IOUParticipantsSplit
- [x] TimezoneSelectPage
- [x] WorkspaceInvitePage
- [x] WorkspaceMembersPage

Beamanator reviewed (approved):
Ok great! Thanks y'all 👍 

@tienifr can you please also add some kind of simple test steps for those pages in the OP as well?

OSBotify commented:
:hand: This PR was not deployed to staging yet because QA is ongoing. It will be automatically deployed to staging after the next production release.

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/4514028469) to staging by https://github.com/Beamanator in version: 1.2.89-0 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅ 
🕸 web 🕸|success ✅

tienifr commented:
@Beamanator Tests updated in the QA section! Everything seems good.  I removed the `YearPickerPage`, since I think that's not relevant to the current issue.

@s77rt Just for confirmation, are you referring to the date picker page in the birthday section? If not, can you provide the location where it is used, because I cannot find a `YearPickerPage` component in the code.

s77rt commented:
@tienifr On birthday page, open date picker, click on year. It will take you to the year picker page which uses OptionsSelector which uses OptionRow.

tienifr commented:
@s77rt Thanks 🎉 I forgot to sync to the newest change

All tests for the related components are added in the QA section. All yours @Beamanator

s77rt commented:
@tienifr This has been merged already.

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/4531239260) to production by https://github.com/luacmartins in version: 1.2.89-0 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅ 
🕸 web 🕸|success ✅
